docusaurus/docs/iOS/guides/working-with-the-composer.md

```diff
@@ -0,0 +1,221 @@
+```swift
+final class iMessageComposerVC: ComposerVC {
+    override func typingMention(in textView: UITextView) -> (String, NSRange)? {
+        // Don't show user suggestions
+        return nil
+    }
+
+    override func typingCommand(in textView: UITextView) -> String? {
+        // Don't show command suggestions
+        return nil
+    }
+}
+
+components.messageComposerVC = iMessageComposerVC.self
+```
+
+## Adding a new Composer State
+
+The `ComposerVC` has by default three different states, `.new`, `.edit` and `.quote`, which are represented by the `ComposerState`. These states define if the composer is creating a new message, editing a new message, or quoting an existing message.
+
+| `.new`  | `.edit` | `.quote` |
+| ------------- | ------------- | ------------- |
+| <img src={require("../assets/composer-ui-state-new.png").default} width="100%"/> | <img src={require("../assets/composer-ui-state-edit.png").default} width="100%"/> | <img src={require("../assets/composer-ui-state-quote.png").default} width="100%"/> |
+
+The `ComposerState` is not enum on purpose so it does not cause breaking changes when adding new states and also it provides the possibility to be extendable by you. You can create a new state by creating an extension:
+
+```swift
+extension ComposerState {
+  public static var custom = ComposerState(rawValue: "custom")
+}
+```
+Then, you can extend `ComposerVC.Content` to support this new state:
+```swift
+extension ComposerVC.Content {
+    public mutating func customState() {
+        self = .init(
+            text: "",
+            state: .custom,
+            editingMessage: nil,
+            quotingMessage: nil,
+            threadMessage: self.threadMessage,
+            attachments: self.attachments,
+            command: nil
+        )
+    }
+}
+```
+For better management of the Composer's state and logic, some of the properties are not mutable, including the `state`, `editingMessage` and `quotingMessage` properties. This is to prevent, for example, the composer content to have an invalid state like having an `editingMessage` and `quotingMessage` at the same time. So, to set a new state you need to re-create the composer content, and if you want some of the data to remain the same, just use the current properties for the new content.
+
+Now, with the new state added, not only you can trigger this new state change by calling `content.customState()`, you can also override the `updateContent()` lifecycle in `ComposerVC` to render this new state:
+```swift
+final class CustomComposerVC: ComposerVC {
+    override func updateContent() {
+        super.updateContent()
+
+        switch content.state {
+        case .custom:
+            composerView.titleLabel.text = "Custom State"
+        default:
+            break
+        }
+    }
+}
```

Comment:
We need to come up with a better example....

Reply:
Yes you are right 🤔 I can't think about a new state actually.. and if even is a valid use case... It is a shame because it was quite the work to document this, but I don't see a point of adding new states at all :/

Reply:
Then let's remove the example or move it to the bottom